internal/csi-addons/rbd/replication.go

```diff
@@ -82,6 +82,12 @@ type ReplicationServer struct {
 	*corerbd.ControllerServer
 }
 
+// NewReplicationServer creates a new ReplicationServer which handles
+// the Replication Service requests from the CSI-Addons specification.
+func NewReplicationServer() *ReplicationServer {
+	return &ReplicationServer{}
+}
```

Comment:
```suggestion
// NewReplicationServer creates a new ReplicationServer which handles
// the Replication Service requests from the CSI-Addons specification.
func NewReplicationServer(c *corerbd.ControllerServer) *ReplicationServer {
	return &ReplicationServer{ControllerServer: c}
}
```

Without the above change, we cannot use the internal locking mechanism. please check this change once again.

Reply:
Thanks for pointing out madhu ! 

volume locks are initialised way later in New[Controller|Node]Server() calls.

We'll need to move this call 

https://github.com/ceph/ceph-csi/blob/306bfa2a35cda92c9a67b5cbbd879e5c32d8708b/internal/rbd/driver/driver.go#L108-L112

to just before this call to make sure everything is initialized.

https://github.com/ceph/ceph-csi/blob/306bfa2a35cda92c9a67b5cbbd879e5c32d8708b/internal/rbd/driver/driver.go#L182-L183

@riya-singhal31 Please make this change too.

Reply:
Updated, thanks!